daemon/cmd/daemon_main.go

```diff
@@ -852,6 +852,12 @@ func initEnv(cmd *cobra.Command) {
 		}
 	}
 
+	// This check is here instead of in DaemonConfig.Populate (invoked at the
+	// start of this function as option.Config.Populate) to avoid an import loop.
+	if option.Config.IdentityAllocationMode == option.IdentityAllocationModeCRD && !k8s.IsEnabled() {
```

Comment:
Looks like we call k8s.IsEnabled() before k8s.Configure()?

I see some other calls in this file to IsEnabled() before Configure(). I wonder if all of them are wrong? Maybe I'm missing something.

Reply:
Joe asked about re practical difference.

IFF KUBERNETES_SERVICE_{HOST,PORT} is not set in the ENV; and
IFF user runs cilium with Kubernetes host set via command line/config; then
AFAIK commit as proposed will declare K8s not configured (even if it was configured).

I think the question is how often we'd have K8s configured via command line/config and not configured via ENV. I don't know answer to this. Maybe it's a moot point given that we already call into IsEnabled() in a few places before we invoke Configure().